src/main/kotlin/org/rust/ide/annotator/fixes/AddStructFieldsFix.kt

```diff
@@ -105,4 +134,58 @@ class AddStructFieldsFix(
 
         return null
     }
+
+    private fun defaultValueExprFor(factory: RsPsiFactory, fieldDecl: RsFieldDecl, ty: Ty): RsExpr {
+        val stdknownItems = StdKnownItems.relativeTo(fieldDecl)
+        return when (ty) {
+            is TyBool -> factory.createExpression("false")
+            is TyInteger -> factory.createExpression("0")
+            is TyFloat -> factory.createExpression("0.0")
+            is TyChar -> factory.createExpression("''")
+            is TyStr -> factory.createExpression("\"\"")
+            is TyReference -> {
+                defaultValueExprFor(factory, fieldDecl, ty.referenced)
+            }
+            is TyEnum -> {
```

Comment:
I don't think we should separate handling of struct and enums. We can manage with single branch `is TyStructOrEnumBase`.